addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1915,6 +1915,48 @@ options.registry.collapse = options.Class.extend({
     },
 });
 
+options.registry.MasonryLayout = options.Class.extend({
+
+    /**
+     * @constructor
+     */
+     init: function () {
+        this._super.apply(this, arguments);
+        this.$container = this.$('>.container, >.container-fluid, >.o_container_small');
+        this.$currentView = this.$container.children();
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Change the masonry snippet layout with templates.
+     */
+    async selectDataAttribute(previewMode, widgetValue, params) {
+        await this._super(...arguments);
+        if (previewMode === 'reset') {
+            this.$container.empty().append(this.$currentView);
+        } else {
+            let layoutPreview = this.$currentView;
+            if (widgetValue !== params.activeValue) {
+                layoutPreview = await this._rpc({
+                    model: 'ir.ui.view',
+                    method: 'render_public_asset',
+                    args: [`${widgetValue}`, {}],
+                    kwargs: {
+                        context: this.options.context,
+                    },
+                });
```

Comment:
A rpc each time we hover a button... meh. And an useless flicker for the duration of the RPC (I am not even testing on runbot).